workflow/validate/lint.go

```diff
@@ -196,3 +138,93 @@ func ParseCWfTmplFromFile(filePath string, strict bool) ([]wfv1.ClusterWorkflowT
 	}
 	return clusterWorkflowTmpls, nil
 }
+
+// Wrapper for collection of all Argo resources parsed from a file / set of files / directory.
+type ArgoResources struct {
+	Workflows                []wfv1.Workflow
+	WorkflowTemplates        []wfv1.WorkflowTemplate
+	CronWorkflows            []wfv1.CronWorkflow
+	ClusterWorkflowTemplates []wfv1.ClusterWorkflowTemplate
+}
+
+func ParseResourcesFromFiles(fileNames []string, strict bool) (*ArgoResources, error) {
```

Comment:
This function could probably be "smarter" and take 2 more `bool` args as inputs, determining whether we even attempt to parse `Workflow` and `CronWorkflow` resources out of the input files (since we never need both for linting, and sometimes don't need either).